src/ProtocolFees.sol

```diff
@@ -44,21 +47,19 @@ abstract contract ProtocolFees is IProtocolFees, Owned {
             assembly {
                 returnData := mload(add(_data, 0x20))
             }
-            // Ensure return data does not overflow a uint16 and that the underlying fees are within bounds.
-            (success, protocolFees) = returnData == uint16(returnData) && _isValidProtocolFee(uint16(returnData))
-                ? (true, uint16(returnData))
+            // Ensure return data does not overflow a uint24 and that the underlying fees are within bounds.
+            (success, protocolFees) = returnData == uint24(returnData) && _isValidProtocolFee(uint24(returnData))
+                ? (true, uint24(returnData))
                 : (false, 0);
         }
     }
 
-    function _isValidProtocolFee(uint16 fee) internal pure returns (bool) {
+    function _isValidProtocolFee(uint24 fee) internal pure returns (bool) {
         if (fee != 0) {
-            uint16 fee0 = fee % 256;
-            uint16 fee1 = fee >> 8;
-            // The fee is specified as a denominator so it cannot be LESS than the MIN_PROTOCOL_FEE_DENOMINATOR (unless it is 0).
-            if (
-                (fee0 != 0 && fee0 < MIN_PROTOCOL_FEE_DENOMINATOR) || (fee1 != 0 && fee1 < MIN_PROTOCOL_FEE_DENOMINATOR)
-            ) {
+            uint16 fee0 = fee.getZeroForOneFee();
+            uint16 fee1 = fee.getOneForZeroFee();
+            // The fee is represented in bips so it cannot be GREATER than the MAX_PROTOCOL_FEE.
+            if ((fee0 > MAX_PROTOCOL_FEE) || (fee1 > MAX_PROTOCOL_FEE)) {
```

Comment:
i'm wondering if we should put this into the protocolFeeLibrary like 
```
return fee0.isValid() && fee1.isValid()
```

and then 

```
function isValid(uint24 self) ... {
   return self > MAX;
}
```

and then we can put MAX_PROTOCOL_FEE into that library instead

Reply:
and then i feel like `BIPS_DENOMINATOR` would ideally be in there too? if theres a neat way to put it in there and not in the pool contract? even if its just then that the pool contract does `ProtocolFeeLibrary.BIPS_DENOMINATOR`

Reply:
I generally feel like what we have now is a pretty good structure to be honest? This function is already supposed to check for validity of the fee so it seems almost overkill to have another library that also checks for validity?

Also in general I think moving things to libraries is good when its reused and we dont ever check validity anywhere else so not really sure its that helpful.

Reply:
I do like the idea though of maybe moving` BPS_DENOMINATOR`. What about moving that to ProtocolFeeLibrary and then in Pool.sol we could have a call thats like `cache.protocolFee.convertToBPS(step.feeAmount`);

Reply:
Or I'd also just be fine leaving as is tbh.. maybe that last line is harder to read than seeing the calculation in Pool.sol